src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/VepTranscriptFmsDTOValidator.java

```diff
@@ -0,0 +1,331 @@
+			for (PredictedVariantConsequence existingPvc : variantLocation.getPredictedVariantConsequences()) {
+				if (transcript.getId() == existingPvc.getVariantTranscript().getId()) {
+					predictedVariantConsequence = existingPvc;
+					isUpdate = true;
+					break;
+				}
+			}
+		}
+		
+		predictedVariantConsequence.setVariantGenomicLocation(variantLocation);
+		predictedVariantConsequence.setVariantTranscript(transcript);
+
+		Map<String, String> attributes = getExtraAttributes(dto);
+		
+		VocabularyTerm vepImpact = null;
+		if (!attributes.containsKey("IMPACT")) {
+			response.addErrorMessage("extra - IMPACT", ValidationConstants.REQUIRED_MESSAGE);
+		} else {
+			vepImpact = vocabularyTermService.getTermInVocabulary(VocabularyConstants.VEP_IMPACT_VOCABULARY, attributes.get("IMPACT")).getEntity();
+			if (vepImpact == null) {
+				response.addErrorMessage("extra - IMPACT", ValidationConstants.INVALID_MESSAGE + " (" + attributes.get("IMPACT") + ")");
+			}
+		}
+		predictedVariantConsequence.setVepImpact(vepImpact);
+		
+		List<SOTerm> vepConsequences = null;
+		if (StringUtils.isBlank(dto.getConsequence())) {
+			response.addErrorMessage("consequence", ValidationConstants.REQUIRED_MESSAGE);
+		} else {
+			vepConsequences = new ArrayList<>();
+			for (String consequence : dto.getConsequence().split(",")) {
+				SearchResponse<SOTerm> soTermResponse = soTermService.findByField("name", consequence);
+				SOTerm vepConsequence = null;
+				if (soTermResponse != null && soTermResponse.getSingleResult() != null
+						&& vocabularyTermService.getTermInVocabulary(VocabularyConstants.VEP_CONSEQUENCE_VOCABULARY, consequence).getEntity() != null) {
+					vepConsequence = soTermResponse.getSingleResult();
+				}
+				if (vepConsequence == null) {
+					response.addErrorMessage("consequence", ValidationConstants.INVALID_MESSAGE + " (" + consequence + ")");
+					break;
+				} else {
+					vepConsequences.add(vepConsequence);
+				}
+			}
+		}
+		predictedVariantConsequence.setVepConsequences(vepConsequences);
+		
+		String hgvsCodingNomenclature = null;
+		if (attributes.containsKey("HGVSc")) {
+			hgvsCodingNomenclature = attributes.get("HGVSc");
+		}
+		predictedVariantConsequence.setHgvsCodingNomenclature(hgvsCodingNomenclature);
+		
+		String hgvsProteinNomenclature = null;
+		if (attributes.containsKey("HGVSp")) {
+			hgvsProteinNomenclature = attributes.get("HGVSp");
+		}
+		predictedVariantConsequence.setHgvsProteinNomenclature(hgvsProteinNomenclature);
+			
+		String referenceCodon = null;
+		String variantCodon = null;
+		if (StringUtils.isNotBlank(dto.getCodons())) {
+			String[] refVarCodons = dto.getCodons().split("/");
+			if (refVarCodons.length == 1 && dto.getConsequence().contains("synonymous_variant")) {
+				referenceCodon = dto.getCodons();
+				variantCodon = dto.getCodons();
+			} else if (refVarCodons.length == 2) {
+				referenceCodon = refVarCodons[0];
+				variantCodon = refVarCodons[1];
+			} else { 
```

Comment:
🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.regexp.RegexpSinglelineJavaCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line must not have trailing whitespace
<!-- __reviewdog__:ChAyYmQzZGQ1MzY1NDE4YTkwEgpjaGVja3N0eWxl -->